IPython/html/files/handlers.py

```diff
@@ -28,12 +28,12 @@ def get(self, path):
         else:
             name = path
         
-        model = cm.get(path)
+        model = cm.get(path, type='file')
         
         if self.get_argument("download", False):
             self.set_header('Content-Disposition','attachment; filename="%s"' % name)
         
-        if model['type'] == 'notebook':
+        if model['type'] == 'notebook' or name.endswith('.ipynb'):
```

Comment:
Should we even be setting an `'application/json'` mimetype here if the data is being morally treated as text?